balancer_conn_wrappers.go

```diff
@@ -159,17 +168,10 @@ func (ccb *ccBalancerWrapper) NewSubConn(addrs []resolver.Address, opts balancer
 }
 
 func (ccb *ccBalancerWrapper) RemoveSubConn(sc balancer.SubConn) {
-	acbw, ok := sc.(*acBalancerWrapper)
-	if !ok {
-		return
-	}
-	ccb.mu.Lock()
-	defer ccb.mu.Unlock()
-	if ccb.subConns == nil {
-		return
-	}
-	delete(ccb.subConns, acbw)
-	ccb.cc.removeAddrConn(acbw.getAddrConn(), errConnDrain)
+	// The RemoveSubConn() is handled in the run() goroutine, to avoid deadlock
```

Comment:
Why is the type assertion removed from here? Was the type assertion present only to make sure that we get a variable of type `*acBalancerWrapper` that we can pass to the `delete(ccb.subConns, acbw)` call, and not to actually prevent any calls to `RemoveSubConn()` with a type other than `*acBalancerWrapper`. 
I'm wondering if this will affect any of our tests which use a `testSubConn`. I guess correctness is not affected, for types other than `*acBalancerWrapper`, we will not match any case statement in `watcher()`. But it is very subtle now, without an explicit type assertion.

Reply:
There's no behavior change before and after. If type is not `*acBalancerWrapper`, we ignore it.

The difference is that we will push non-`*acBalancerWrapper` to the queue, so it may cause some unnecessary handling.
I can do the type assertion here, and not `put()` for the other types.
But then, we will still need to type assert in the goroutine again, so two type asserts for every `*acBalancerWrapper`?

> our tests which use a testSubConn
With testSubConn, we also use a testClientConn. So it never touches this code.
And for more realistic tests, with real ClientConn, we call `NewSubConn()` to make the SubConn.